### PR TITLE
SM-837 SM input names validation

### PR DESCRIPTION
**Pull request by mzieniukbw**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/SM-837

## 📔 Objective

Input validation and sanitisation for SM should be the same as we do in `bitwarden/clients`

Validation:
- Project create or update on fields:
  - `name`: required, max length 500
- Secret create or update on fields:
  - `key`: required, max length 500
  - `value`: required, max length 25000
  - `notes`: max length 7000

Sanitisation to SM cli:
- Project create or update on fields:
  - `name`: trim
- Secret create or update on fields:
  - `key`: trim
  - `notes`: trim

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation
  team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by mzieniukbw** [comment]
I have made a mistake with the ticket no, it should be SM-837, not SM-836, might need to re-do the PR or force push 😅

**Review by cd-bitwarden — COMMENTED** [review]
This looks great 
Thank you!

I am not a pro at rust yet, so I do recommend @coltonhurst also review this if he has time, but from what I can tell it looks good 😄

**Review by coltonhurst — COMMENTED** [review]
Thanks for these changes @mzieniukbw 😊

As we discussed internally we might not need the whole library, we could keep a version of the error type you created and do more simple validation checks for now until we need something bigger.

But would love thoughts from @dani-garcia & @Hinton in case this could be used for more?

**Review by dani-garcia — COMMENTED** [review]
Looks pretty good, some minor comments.

I'm okay with using the library, you're right that we're not doing much validation at the moment, but I think it would be good to try it out.

File: crates/bws/Cargo.toml
Comment (mzieniukbw) [reply crates/bws/Cargo.toml]: Fixed in https://github.com/bitwarden/sdk/pull/810/commits/e5b09820e872a261bac44f3802711205b4a5bede

File: crates/bitwarden/src/error.rs
Comment (mzieniukbw) [reply crates/bitwarden/src/error.rs]: Fixed in https://github.com/bitwarden/sdk/pull/810/commits/b1651e3ab9dd9ad996c8618aeaeb12aab8efa54f

File: crates/bitwarden/src/secrets_manager/projects/create.rs
Comment (mzieniukbw) [reply crates/bitwarden/src/secrets_manager/projects/create.rs]: Fixed in https://github.com/bitwarden/sdk/pull/810/commits/b9b0c61d85ba2ff6054d3581b54b0fe09ca79175

File: crates/bitwarden/src/secrets_manager/projects/create.rs
Comment (mzieniukbw) [reply crates/bitwarden/src/secrets_manager/projects/create.rs]: Fixed in https://github.com/bitwarden/sdk/pull/810/commits/576728c5822ddfd865fd37c573d896c100f8e438

File: crates/bitwarden/src/error.rs
Comment (mzieniukbw) [reply crates/bitwarden/src/error.rs]: Fixed in https://github.com/bitwarden/sdk/pull/810/commits/2419144668ec09097cd009d3f0876442efb0a507

**Review by dani-garcia — APPROVED** [review]

**Review by coltonhurst — COMMENTED** [review]
< removed my comments, please disregard >

**Review by coltonhurst — DISMISSED** [review]

**Comment by mzieniukbw** [comment]
@coltonhurst i had to resolve a merge conflict before merging, please approve again if that's ok.

**Review by dani-garcia — APPROVED** [review]

**Review by coltonhurst — APPROVED** [review]

**Pull request merged**
